src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -182,11 +199,31 @@ public async Task<bool> TryRefreshAsync()
             {
                 await RefreshAsync().ConfigureAwait(false);
             }
-            catch (Exception e) when (
-                e is KeyVaultReferenceException ||
-                e is RequestFailedException ||
-                ((e as AggregateException)?.InnerExceptions?.All(e => e is RequestFailedException) ?? false) ||
-                e is OperationCanceledException)
+            catch (RequestFailedException e)
+            {
+                if (e.Status == (int)HttpStatusCode.Unauthorized || e.Status == (int)HttpStatusCode.Forbidden)
+                {
+                    _logger?.LogError(e, "Refresh operation failed due to an authentication error.");
+                }

```

Comment:
Should we log something in the `else` or we are missing a failure case here?

Reply:
I think the [original plan](https://github.com/Azure/AppConfiguration-DotnetProvider/issues/113#issuecomment-582167408) was to log only authentication/Key Vault errors for which users might need to take some action. Everything else is transient and might be resolved with the next refresh operation.

Reply:
Yes, we agreed everything could be transient including auth issues. Customers want to know when "transient" errors happened so they can react if necessary. We log details when we know better, but we should still log when we don't have details.

Reply:
Added a warning for non-auth issues too.